pkg/install.go

```diff
@@ -253,51 +247,43 @@ func InstallGoChaosEngine(testsDetails *types.TestDetails, chaosEngine *types.Ch
 
 	// for ec2-terminate instance
 	if testsDetails.ExperimentName == "ec2-terminate" {
-		environments["EC2_INSTANCE_ID"] = testsDetails.InstanceID
-		environments["REGION"] = testsDetails.Region
+		envDetails.SetEnv("EC2_INSTANCE_ID", testsDetails.InstanceID).
+			SetEnv("REGION", testsDetails.Region)
 	}
 
 	// for experiments like pod network latency
-	if testsDetails.NetworkLatency != "" {
-		environments["NETWORK_LATENCY"] = testsDetails.NetworkLatency
-	}
+	envDetails.SetEnv("NETWORK_LATENCY", testsDetails.NetworkLatency)
 
 	// update Engine if FillPercentage if it does not match criteria
 	if testsDetails.FillPercentage != 80 {
-		environments["FILL_PERCENTAGE"] = strconv.Itoa(testsDetails.FillPercentage)
+		envDetails.SetEnv("FILL_PERCENTAGE", strconv.Itoa(testsDetails.FillPercentage))
 	}
 
 	// update App Node Details
 	if testsDetails.ApplicationNodeName != "" {
-		environments["TARGET_NODE"] = testsDetails.ApplicationNodeName
-		chaosEngine.Spec.Experiments[0].Spec.Components.NodeSelector.KubernetesIoHostname = testsDetails.NodeSelectorName
+		envDetails.SetEnv("TARGET_NODE", testsDetails.ApplicationNodeName)
+		chaosEngine.Spec.Experiments[0].Spec.Components.NodeSelector["kubernetes.io/hostname"] = testsDetails.NodeSelectorName
```

Comment:
can you check the engine creation for kubelet, node-drain, node-taint. https://github.com/litmuschaos/litmus-e2e/runs/2713155334?check_suite_focus=true

Reply:
We test these experiments in kubeadm or any other cluster where the DNS name and node names are the same.